taskcluster/scopes.rst

```diff
@@ -0,0 +1,69 @@
+
+    queue:get-artifact:releng/super-sekrit/*
+
+rather than::
+
+    queue:get-artifact:releng/super-sekrit*
+
+Ci-configuration
+----------------
+
+We grant scopes to clients and roles. These are defined in the `ci-configuration repo <https://hg.mozilla.org/ci/ci-configuration/>`__; view the `README <https://hg.mozilla.org/ci/ci-configuration/file/tip/README.md>`__.
+
+We test, diff, and apply these configuration changes using the `ci-admin repo <https://hg.mozilla.org/ci/ci-admin/>`__; view the `README <https://hg.mozilla.org/ci/ci-admin/file/tip/README.md>`__.
+
+Conventions
+-----------
+
+Groups / teams
+~~~~~~~~~~~~~~
+We try to tie most user scope grants to LDAP. Grants to ``mozilla-group::GROUP`` will assign the scopes to users that belong to that MoCo ldap group. Grants to ``mozillians-group::GROUP`` will grant scopes to users that belong to that Mozillians group (`people.mozilla.org <https://people.mozilla.org>`__).
+
+We also define ``ci-group`` roles like ``project:releng:ci-group:team_moco`` in `this block <https://hg.mozilla.org/ci/ci-configuration/file/307d8717f17e3916ebdfc54e58705230c5cf30a7/grants.yml#l2351>`__.
+
+Levels
+~~~~~~
+
+Levels in scopes match the Firefox commit levels. Level 1 is Try and pull requests; contributors can easily get this level of access. Level 2 is projects and l10n, and isn't used everywhere. Level 3 is release level, and requires a higher bar to gain this level of access. Ideally contributors will be able to get everything done at level 1 unless they become a trusted member of a project.
+
+We encode levels in workerType/workerPool names, and in other scopes that should be restricted by repo and commit level. For example, the ``gecko-1/decision`` worker is the decision worker for Try. ``gecko-3/decision`` is the trusted decision worker for release trains and autoland.
+
+Docker- and Generic-Worker scopes
+---------------------------------
+
+The workers for docker- and generic-worker should be minimal, just enough to register as a given workerType and claim tasks from the queue. They will be granted temporary scopes for each task that they run.
+
+Scriptworker scopes
+-------------------
+
+Scriptworker scopes are similar, but each ``*script`` will also define script-specific scopes, like ``project:releng:signing:format:signcode``.
+
+In addition, until we fix `Issue #426 (use temp queue to download artifacts) <https://github.com/mozilla-releng/scriptworker/issues/426>`__, we also need to grant private artifact scopes to the *clientId* as well as the task.
+
+Restricted scopes
+~~~~~~~~~~~~~~~~~
+
+We define `cot_restricted_scopes <https://github.com/mozilla-releng/scriptworker/blob/dd0eed21354ecfabbe5838ea3cf730ff0630a3dd/src/scriptworker/constants.py#L361-L445>`__ in scriptworker. These are scopes that can only run on specific allowlisted trees or ``tasks_for``.
```

Comment:
As mentioned this is the initial document so I don't expect it to be complete but it does read more like a glossary. How and why would we expect someone to use this document?

It would be good to include a list of common tasks and more "how to" information on creating scopes perhaps as part of a different document.

I checked the links and they are all good but it will be nice to see this rendered in markdown to double check.

Minor nit for the workers/scopes edit but seems fine. Were you expecting more from me on this ?

I suppose turn about is fair play after I dumped all those reviews on you! ;-)

Reply:
PS. aerickson has been involved with taskcluster clients and scopes more recently that myself. He might be a good sounding board as well.

Reply:
Yeah, I definitely realize that this doc is super skeleton-y and could use a lot of improvement and fleshing out.
I just wanted to have an initial doc about scopes that we can add to and update as things change: it's easier for someone to amend an existing doc than to create one from scratch.

I think I sent the review to `mozilla/releng` as a group review :) Thanks for jumping on this.